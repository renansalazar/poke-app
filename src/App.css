body{
  width: 100%;
  height: 100%;
}

*{
  box-sizing: border-box;
}

header{
  background: #0bcdff;
  background: red;
  padding: 10px; 
  display: flex;
  justify-content: center;
}

header a{
  color: rgba(255, 255, 0, 0.918);
  font-size: 23px;
  text-decoration: none;
  transition: all 200ms;
}

header a:hover{
  color: #0bcdff;
}

.container{
  margin: 20px;
}

.containerLista{
  margin-top: 10px;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 6px;
}

.containerItem{
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-transform: capitalize;
  background: #0bcdff;
  border-radius: 10px;
  border: 1px solid gray;
  margin: 20px;
  transition: all 200ms;
}

.containerItem:hover{
  box-shadow: 10px 10px 10px gray;
}

.containerLista a{
  color: white;
  text-decoration: none;
  text-shadow: 1px 1px 1px black;
}

.typeItem{
  margin: auto 6px;
  background: orange;
  border-radius: 6px;
  padding: 6px;
  display: flex;
  align-items: center;
}

.containerSprite{
  position: relative;
  height: 260px;
  background-color: yellow;
  width: 260px;
  padding: 20px;
  margin: 20px;
  border-radius: 16px;
  overflow: hidden;
}

.containerSprite img{
  background-color: white;
  position: absolute;
  width: 220px;
  height: 220px;
  margin: auto;
}

.containerSprite img:nth-child(1){
  z-index: 1;
  opacity: 1;
  animation: fadeinout 4s infinite backwards;
}

@keyframes fadeinout {
  0%,49% { opacity: 0; }
  50%,100% { opacity: 1; }
}

.containerBuscador {
  display: flex;
  justify-content: center;
}

.containerBuscador input[type=search],
.containerBuscador button{
  padding: 10px;
  font-size: 16px;
  height: 50px;
}


.containerBuscador input[type=search]{
  border: 1px solid gray;
  background: black;
  color: yellow;
  border-radius: 6px 0 0 6px;
}

.containerBuscador button{
  border: 1px solid gray;
  border-radius: 0 6px 6px 0 ;
  cursor: pointer;
  background: #0bcdff;
}

.containerButtonLoadMore{
  display: flex;
  justify-content: center;
}

.buttonLoadMore{
  margin: 0 auto;
  padding: 10px;
  background: #0bcdff;
  border-radius: 6px;
  cursor: pointer;
  transition: all 200ms;
}

.buttonLoadMore:hover{
  background: gray;
  color: white;
}

.pokedex{
  max-width: 900px;
  margin: auto;
  display: grid;
  grid-template-columns: 1fr 1fr;
}

.pokedex .first{
  background: red;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  border-radius: 50px 0 0 50px;
  padding: 20px;
}

.joystick{
  width: 100%;
  padding: 20px;
  display: flex;
  justify-content: space-between;
}

.pokedex .first .button{
  background: black;
  border-radius: 50%;
  width: 80px;
  height: 80px;
}

.pokedex .first .cross{
  width: 100px;
  height: 100px;
  background: black;
  clip-path: polygon(0 38%, 38% 38%, 38% 0%, 64% 0%, 64% 38%, 100% 38%, 
  100% 64%, 64% 64%, 64% 100%, 38% 100%, 38% 64%, 0 64%);
}

.pokedex .second {
  background-color: red;
  border-radius: 0 0 50px 0;
}
.pokedex .second .console{
  margin: 20px;
  padding: 20px;
  border-radius: 10px;
  background-color: black;
  color: white;
  display: flex;
  flex-direction: column;
}

.pokedex .second .containerTypes{
  display: flex;
}

.pokedex .second .stats{
  display: grid;
  grid-auto-columns: 1fr;
  grid-auto-flow: column;
}

.pokedex .second .stats>div{
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: center;
}

.pokedex .second .stats .bar{
  margin-top: 2px;
  width: 20px;
  height: 1px;
  display: block;
  background-color: yellow;
  content: '';
}

.pokedex .second .stats>div .statName{
  font-size: .9rem;
  height: 50px;
}

.containerTriaje{
  display: flex;
  justify-content: space-around;
}

.containerTriaje .console{
  flex-direction: row!important;
  align-items: center;
  gap: 6px;
  color: yellow!important;
}

.containerTriaje .console h5{
  margin: 0;
}

.containerName{
  position: absolute;
  color: black;
  bottom: 18px;
  z-index: 9;
  text-transform: capitalize;
  width: 220px;
  font-weight: bold;
  text-align: center;
}