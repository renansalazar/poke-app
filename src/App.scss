body{
  width: 100%;
  height: 100%;
}

*{
  box-sizing: border-box;
}

.container{
  margin: 20px;
}

header{
  background: #0bcdff;
  background: red;
  padding: 10px; 
  display: flex;
  justify-content: center;
  a{
    color: rgba(255, 255, 0, 0.918);
    font-size: 27px;
    text-decoration: none;
    transition: all 200ms;
    &:hover{
      color: #0bcdff;
    }
  }
  
}
.containerLista{
  .containerLista__content{
    margin-top: 10px;
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    grid-gap: 6px;
    
    a{
      color: white;
      text-decoration: none;
      text-shadow: 1px 1px 1px black;
    }
    
  }
  
  .containerLista__wrapperButton{
    display: flex;
    justify-content: center;
  }
}
  
.containerItem{
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-transform: capitalize;
  background: #0bcdff;
  border-radius: 10px;
  border: 1px solid gray;
  margin: 20px;
  transition: all 200ms;
  &:hover{
    box-shadow: 10px 10px 10px gray;
  }
}


.containerBuscador {
  display: flex;
  justify-content: center;
  input[type=search],
  button{
    padding: 10px;
    font-size: 16px;
    height: 50px;
  }
}



.containerBuscador input[type=search]{
  border: 1px solid gray;
  background: black;
  color: yellow;
  border-radius: 6px 0 0 6px;
}

.containerBuscador button{
  border: 1px solid gray;
  border-radius: 0 6px 6px 0 ;
  cursor: pointer;
  background: #0bcdff;
}

.containerLista__wrapperButton__button{
  margin: 0 auto;
  padding: 10px;
  background: #0bcdff;
  border-radius: 6px;
  cursor: pointer;
  transition: all 200ms;
}

.containerLista__wrapperButton__button:hover{
  background: gray;
  color: white;
}