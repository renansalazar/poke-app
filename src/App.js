import { BrowserRouter, Routes, Route } from "react-router-dom"

import './App.css'
import Detalle from "./pages/Detalle"
import Home from './pages/Home'
import Layout from "./pages/Layout"
import NoMatch from "./pages/NoMatch"
import { QueryClientProvider, QueryClient } from 'react-query'
import { ReactQueryDevtools } from 'react-query/devtools'

const queryClient = new QueryClient()
function App() {
  return (
  <QueryClientProvider client={queryClient}>
    <BrowserRouter>
      <Routes>
          <Route  path="/" element={<Layout />}>
              <Route index element={<Home />} />
              <Route path="/detalle/:name" element={<Detalle />} />
              <Route path="*" element={<NoMatch />} />
          </Route>
      </Routes>
    </BrowserRouter>
    <ReactQueryDevtools initialIsOpen={false} />
  </QueryClientProvider>
  )
}

export default App;
