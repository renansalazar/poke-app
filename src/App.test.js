import React from 'react';
import {render} from '@testing-library/react';
import  App from './App';

describe('App component', () => {
  test('app test', () => {
    const { container } = render(
          <App />
    )
    expect(container.firstChild).toMatchSnapshot()
  });
});