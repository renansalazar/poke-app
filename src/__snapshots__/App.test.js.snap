// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component app test 1`] = `
<header>
  <a
    href="/"
  >
    Pokedex
  </a>
</header>
`;
